src/osdatahub/LinkedIdentifiersAPI/linked_identifiers_api.py

```diff
@@ -0,0 +1,97 @@
+from typing import Union
+
+import requests
+from osdatahub.errors import raise_http_error
+from osdatahub.LinkedIdentifiersAPI.linked_identifier_options import (
+    correlation_methods, feature_types, identifier_types)
+from typeguard import typechecked
+
+
+class LinkedIdentifiersAPI:
+    """Main class for querying the OS Linked Identifiers API (https://osdatahub.os.uk/docs/linkedIdentifiers/overview)
+
+    Args:
+        key (str): A valid OS API Key. Get a free key here - https://osdatahub.os.uk/
+
+    Example::
+
+        from osdatahub import LinkedIdentifiersAPI
+        from os import environ
+
+        key = environ.get("OS_API_KEY")
+        linked_ids = LinkedIdentifiersAPI(key)
+        results = linked_ids.query(200001025758)
+    """
+
+    __ENDPOINT = r"https://api.os.uk/search/links/v1/"
+
+    def __init__(self, key: str):
+        self.key = key
+
+    def __request(self, endpoint: str) -> dict:
+        response = requests.get(endpoint)
+        if response.status_code != 200:
+            raise_http_error(response)
+        return response.json()
+
+    def __get_endpoint(self, id: Union[int, str],
+                       feature_type: str, identifier_type: str) -> str:
+        if feature_type is not None and identifier_type is not None:
+            raise ValueError("It is possible to query by the feature_type " +\
+                             "OR the identifier type, but not both")
+        elif feature_type is not None:
+            feature_types.validate(feature_type)
+            query = f"featureTypes/{feature_type}/{id}?key={self.key}"
+        elif identifier_type is not None:
+            identifier_types.validate(identifier_type)
+            query = f"identifierTypes/{identifier_type}/{id}?key={self.key}"
+        else:
+            query = f"identifiers/{id}?key={self.key}"
+        return self.__ENDPOINT + query
+
+    @typechecked
+    def query(self, id: Union[int, str],
```

Comment:
Consider using something other than id - it is the name of a built-in function

Reply:
renamed to identifier